external/pdal_wrench/to_raster_tin.cpp

```diff
@@ -0,0 +1,259 @@
+                copc_opts.add(pdal::Option("bounds", box_to_pdal_bounds(tile->boxWithCollar)));
+                reader->addOptions(copc_opts);
+            }
+        }
+    }
+
+    Stage &delaunay = manager->makeFilter("filters.delaunay");
+    for (Stage *stage : readers)
+    {
+        delaunay.setInput(*stage);  // connect all readers to the writer
+    }
+
+    if (!tile->filterExpression.empty())
+    {
+        Options filter_opts;
+        filter_opts.add(pdal::Option("where", tile->filterExpression));
+        delaunay.addOptions(filter_opts);
+    }
+
+    pdal::Options faceRaster_opts;
+    faceRaster_opts.add(pdal::Option("resolution", resolution));
+
+    if (tile->box.valid())  // if box is not provided, filters.faceraster will calculate it from data
+    {
+        faceRaster_opts.add(pdal::Option("origin_x", tile->box.minx));
+        faceRaster_opts.add(pdal::Option("origin_y", tile->box.miny));
+        faceRaster_opts.add(pdal::Option("width", (tile->box.maxx-tile->box.minx)/resolution));
+        faceRaster_opts.add(pdal::Option("height", (tile->box.maxy-tile->box.miny)/resolution));
+    }
+
+    Stage &faceRaster = manager->makeFilter("filters.faceraster", delaunay, faceRaster_opts);
+
+    pdal::Options writer_opts;
+    writer_opts.add(pdal::Option("data_type", "float32"));  // default was float64 which seems like too much
+    writer_opts.add(pdal::Option("gdalopts", "TILED=YES"));
+    writer_opts.add(pdal::Option("gdalopts", "COMPRESS=DEFLATE"));
+    Stage &w = manager->makeWriter(tile->outputFilename, "writers.raster", faceRaster);
+
+    return manager;
+}
+
+
+void ToRasterTin::preparePipelines(std::vector<std::unique_ptr<PipelineManager>>& pipelines, const BOX3D &bounds, point_count_t &totalPoints)
+{
+    if (ends_with(inputFile, ".vpc"))
+    {
+        // using spatial processing
+
+        VirtualPointCloud vpc;
+        if (!vpc.read(inputFile))
+            return;
+
+        // for /tmp/hello.tif we will use /tmp/hello dir for all results
+        fs::path outputParentDir = fs::path(outputFile).parent_path();
+        fs::path outputSubdir = outputParentDir / fs::path(outputFile).stem();
+        fs::create_directories(outputSubdir);
+
+        // TODO: optionally adjust origin to have nicer numbers for bounds?
+        if (tileAlignment.originX == -1)
+            tileAlignment.originX = bounds.minx;
+        if (tileAlignment.originY == -1)
+            tileAlignment.originY = bounds.miny;
+
+        // align bounding box of data to the grid
+        TileAlignment gridAlignment = tileAlignment;
+        gridAlignment.tileSize = resolution;
+        Tiling gridTiling = gridAlignment.coverBounds(bounds.to2d());
+        std::cout << "grid " << gridTiling.tileCountX << "x" << gridTiling.tileCountY << std::endl;
+        BOX2D gridBounds = gridTiling.fullBox();
+
+        Tiling t = tileAlignment.coverBounds(gridBounds);
+        std::cout << "tiles " << t.tileCountX << " " << t.tileCountY << std::endl;
```

Comment:
can all this output code be disabled? (at least in release builds)

Reply:
Hidden behind verbose flag in https://github.com/PDAL/wrench/commit/b27ebbd0421ee027fe60eea746565e0932805c24